src/NuGet.Core/NuGet.ProjectModel/LockFile/Utf8JsonStreamLockFileTargetLibraryConverter.cs

```diff
@@ -0,0 +1,159 @@
+                    lockFileTargetLibrary.NativeLibraries = reader.ReadObjectAsList<LockFileItem>(Utf8JsonReaderExtensions.LockFileItemConverter);
+                }
+                else if (reader.ValueTextEquals(BuildPropertyName))
+                {
+                    reader.Read();
+                    lockFileTargetLibrary.Build = reader.ReadObjectAsList<LockFileItem>(Utf8JsonReaderExtensions.LockFileItemConverter);
+                }
+                else if (reader.ValueTextEquals(BuildMultiTargetingPropertyName))
+                {
+                    reader.Read();
+                    lockFileTargetLibrary.BuildMultiTargeting = reader.ReadObjectAsList<LockFileItem>(Utf8JsonReaderExtensions.LockFileItemConverter);
+                }
+                else if (reader.ValueTextEquals(ContentFilesPropertyName))
+                {
+                    reader.Read();
+                    lockFileTargetLibrary.ContentFiles = reader.ReadObjectAsList<LockFileContentFile>(Utf8JsonReaderExtensions.LockFileContentFileConverter);
+                }
+                else if (reader.ValueTextEquals(RuntimeTargetsPropertyName))
+                {
+                    reader.Read();
+                    lockFileTargetLibrary.RuntimeTargets = reader.ReadObjectAsList<LockFileRuntimeTarget>(Utf8JsonReaderExtensions.LockFileRuntimeTargetConverter);
+                }
+                else if (reader.ValueTextEquals(ToolsPropertyName))
+                {
+                    reader.Read();
+                    lockFileTargetLibrary.ToolsAssemblies = reader.ReadObjectAsList<LockFileItem>(Utf8JsonReaderExtensions.LockFileItemConverter);
+                }
+                else if (reader.ValueTextEquals(EmbedPropertyName))
+                {
+                    reader.Read();
+                    lockFileTargetLibrary.EmbedAssemblies = reader.ReadObjectAsList<LockFileItem>(Utf8JsonReaderExtensions.LockFileItemConverter);
+                }
+                else if (reader.ValueTextEquals(FrameworkReferencesPropertyName))
+                {
+                    reader.Read();
+                    lockFileTargetLibrary.FrameworkReferences = reader.ReadStringArrayAsIList();
+                }
+                else
+                {
+                    reader.Skip();
+                }
+            }
+            lockFileTargetLibrary.Freeze();
+            return lockFileTargetLibrary;
+        }
+
+        private IList<PackageDependency> ReadPackageDependencyList(ref Utf8JsonStreamReader reader)
+        {
+            if (reader.TokenType != JsonTokenType.StartObject)
+            {
+                return Array.Empty<PackageDependency>();
+            }
+
+            var packageDependencies = new List<PackageDependency>();
+            while (reader.Read() && reader.TokenType == JsonTokenType.PropertyName)
+            {
+                string propertyName = reader.GetString();
+                string versionString = reader.ReadNextTokenAsString();
+
```

Comment:
35% of allocations come from these values, I feel like sharing the strings for common dependencies would be helpful in reducing allocations, especially given the version number is immediately thrown away.